js_modules/dagit/packages/core/src/instance/BackfillTable.tsx

```diff
@@ -0,0 +1,555 @@
+import {gql, useMutation} from '@apollo/client';
+import {
+  Box,
+  Button,
+  Colors,
+  Group,
+  Icon,
+  MenuItem,
+  Menu,
+  Popover,
+  Table,
+  Tag,
+  Mono,
+  stringFromValue,
+} from '@dagster-io/ui';
+import qs from 'qs';
+import * as React from 'react';
+import {useHistory, Link} from 'react-router-dom';
+import styled from 'styled-components/macro';
+
+import {showCustomAlert} from '../app/CustomAlertProvider';
+import {SharedToaster} from '../app/DomUtils';
+import {usePermissions} from '../app/Permissions';
+import {PythonErrorInfo, PYTHON_ERROR_FRAGMENT} from '../app/PythonErrorInfo';
+import {PartitionStatus} from '../partitions/PartitionStatus';
+import {PipelineReference} from '../pipelines/PipelineReference';
+import {
+  doneStatuses,
+  failedStatuses,
+  inProgressStatuses,
+  queuedStatuses,
+  successStatuses,
+} from '../runs/RunStatuses';
+import {DagsterTag} from '../runs/RunTag';
+import {runsPathWithFilters} from '../runs/RunsFilterInput';
+import {TerminationDialog} from '../runs/TerminationDialog';
+import {TimestampDisplay} from '../schedules/TimestampDisplay';
+import {BulkActionStatus, RunStatus} from '../types/globalTypes';
+import {isThisThingAJob, useRepository} from '../workspace/WorkspaceContext';
+import {buildRepoAddress} from '../workspace/buildRepoAddress';
+import {repoAddressAsString} from '../workspace/repoAddressAsString';
+import {workspacePathFromAddress, workspacePipelinePath} from '../workspace/workspacePath';
+
+import {BackfillPartitionsRequestedDialog} from './BackfillPartitionsRequestedDialog';
+import {BackfillStepStatusDialog} from './BackfillStepStatusDialog';
+import {BackfillTerminationDialog} from './BackfillTerminationDialog';
+import {RESUME_BACKFILL_MUTATION} from './BackfillUtils';
+import {
+  BackfillTableFragment,
+  BackfillTableFragment_runs,
+  BackfillTableFragment_partitionSet,
+} from './types/BackfillTableFragment';
+import {resumeBackfill, resumeBackfillVariables} from './types/resumeBackfill';
+
+type BackfillRun = BackfillTableFragment_runs;
+
+export const BackfillTable = ({
+  showPartitionSet = true,
+  allPartitions,
+  backfills,
+  refetch,
+}: {
+  allPartitions?: string[];
+  backfills: BackfillTableFragment[];
+  refetch: () => void;
+  showPartitionSet?: boolean;
+}) => {
+  const [terminationBackfill, setTerminationBackfill] = React.useState<BackfillTableFragment>();
+  const [stepStatusBackfill, setStepStatusBackfill] = React.useState<BackfillTableFragment>();
+  const [
+    partitionsRequestedBackfill,
+    setPartitionsRequestedBackfill,
+  ] = React.useState<BackfillTableFragment>();
+  const [resumeBackfill] = useMutation<resumeBackfill, resumeBackfillVariables>(
+    RESUME_BACKFILL_MUTATION,
+  );
+  const [cancelRunBackfill, setCancelRunBackfill] = React.useState<BackfillTableFragment>();
```

Comment:
I think this is good but if we end up creating any more modals, one trick we've used elsewhere is to define a union type like `ActiveModal = {type: 'termination', items: BackfillTableFragment} | {type: 'stepStatus', items: BackfillTableFragment}...` so that these can all be placed in the same state.